app/controllers/application_controller/tags.rb

```diff
@@ -201,9 +201,13 @@ def tagging_save_tags
   def tag_edit_build_screen
     @showlinks = true
 
-    cats = Classification.categories.select(&:show).sort_by { |t| t.description.try(:downcase) } # Get the categories, sort by description
+    cats = Classification.categories.select(&:show) # Get the categories
     @categories = {}    # Classifications array for first chooser
+
+    mapped_categories = cats.select { |c| ContainerLabelTagMapping::TAG_PREFIXES.any? { |pattern| c.tag.name.include?(pattern) } }
```

Comment:
I'd like `TAG_PREFIXES` to remain an implementation detail internal to the model.
I did this kludge after the "correct" join, through parent category, to ContainerLabelTagMapping proved too complicated to debug, but in any case my hope was to encapsulate that in `ContainerLabelTagMapping.controls_tag?` and `Tag.controlled_by_mapping` scope.
If these are not sufficient, perhaps we need more abstraction eg. adding `Classification.controlled_by_mapping` but let's not leak TAG_PREFIXES...